snyk/cli
# Feat: Implement IaC output failures list section [CFG-1572]

#### What does this PR do?

- Implements the new failures list section for the Snyk IaC test output.
- Applies the new failures list section in the test flow, wrapped in the `iacCliOutput` feature flag.

#### Notes for the reviewer

- The relatively high number of files is mainly due to the added fixture files for the tests.
- As in the original output, for flows in which the output only consists of a single error message, the case is the same for this implementation of the new output as well as of now. Other output design options are currently being explored for this single error message output ([see here](https://snyk.slack.com/archives/C030UA3U1RB/p1649271803177019))

#### Where should the reviewer start?

- `src/cli/commands/test/iac/index.ts`

#### How should this be manually tested?

- Execute the `snyk iac test` command without the `iacCliOutput` feature flag, and ensure the output is the same as the original.
- Execute the `snyk iac test` command with the `iacCliOutput` feature flag, ensure the test failures are displayed in the new section, and that it matches [the design](https://www.figma.com/file/buNFZqr1Uhf4SaifHw4WXE/CLI-IaC-Output-Improvements?node-id=0%3A1).

#### Any background context you want to provide?

- As part of the new design for the Snyk IaC test output , we’d like to replace the current per-project failure reasons output, with an aggregated failures summary at the end of the test output. This can be viewed in [the following design](https://www.figma.com/file/buNFZqr1Uhf4SaifHw4WXE/CLI-IaC-Output-Improvements?node-id=0%3A1).

#### What are the relevant tickets?

- [CFG-1572](https://snyksec.atlassian.net/browse/CFG-1572)

#### Screenshots

- Before
    ![image](https://user-images.githubusercontent.com/46415136/163410949-18cc3968-23c8-41a9-b1c9-e74bec010544.png)

- After
    ![image](https://user-images.githubusercontent.com/46415136/163410827-8a26b26b-ebd4-4ee9-978b-5bdf673bc1ae.png)


#### Additional questions
